src/dmd/parse.d

```diff
@@ -247,9 +249,44 @@ final class Parser(AST) : Lexer
             /* Create a pseudo-filename for the mixin string, as it may not even exist
              * in the source file.
              */
-            char* filename = cast(char*)mem.xmalloc(strlen(loc.filename) + 7 + (loc.linnum).sizeof * 3 + 1);
-            sprintf(filename, "%s-mixin-%d", loc.filename, cast(int)loc.linnum);
-            scanloc.filename = filename;
```

Comment:
Oh yeah, it did do this didn't it. This is why I've always been so annoyed ;)
Ideally, for cases where not `-debugmixin`, it would set the file/line to point right at the mixin statement, rather than hassling about an artificial filename that doesn't exist.

Reply:
Yes, thats why i was a little confused. That should be doable.

Reply:
I'd consider it a stretch goal ;) .. There's no actual regression if we don't get this in there.

Reply:
Nobody will notice, because as soon as `-debugmixin` exists, there would be no reason not to use it whenever you use `-g`.

Reply:
I'll make `-g` imply `-debugmixins`. The reverse in not necessarily true. e.g. when trying to decipher why a doubly (or more) nested mixin does not compile one does not need debug info one needs source, so this will remain a separate flag.

Reply:
While I find it agreeable, I'm not sure that's a good idea. I think you'll receive strong resistance.
I have a sense that the compiler probably shouldn't write any more output than it was told to...